src/avatar/avatar.component.ts

```diff
@@ -84,11 +85,12 @@ export class IgxAvatar {
     }
 
     public ngAfterViewInit() {
-        if (this.initials && this.image) {
-            const src = this.generateInitials(
-                parseInt(this.image.nativeElement.width, 10)
-            );
-            this.image.nativeElement.src = src;
+        if (this.initials && this.initialsImage) {
+            // it seems the svg element is not yet fully initialized so give it some time
+            setTimeout(() => {
+                const size = parseInt(this.initialsImage.nativeElement.width.baseVal.value, 10);
```

Comment:
Use Renderer2 to modify attributes on the element. This will allow server side rendering and will work in web workers too, where the DOM is not available.